Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -18,28 +25,114 @@ def init_locals(WorkspaceId, RegDomain):
 def Set_Marshall(WorkspaceId, Enabled, RegDomain):
     (WorkspaceId, RegDomain) = init_locals(WorkspaceId, RegDomain)
     if (Enabled):
-        i = 1
+        #call the registration script
+        try:
+            agent_id = read_oms_config_file()
+            sp = subprocess.call(
+                ["python", REGISTRATION_FILE_PATH, "--register", "-w " + WorkspaceId, "-a " + agent_id,
+                 "-c " + OMS_CERTIFICATE_PATH, "-k " + OMS_CERT_KEY_PATH, "-p " + WORKING_DIRECTORY_PATH])
+            if sp != 0:
+                LG().Log('ERROR', "Linux Hybrid Worker registration failed")
```

Comment:
Please add the output of STDERR if the process exits with something else than 0 to the log (will help us debug).